src/foam/nanos/auth/CachingAuthService.java

```diff
@@ -69,18 +72,28 @@ public CachingAuthService(AuthService delegate, String[] extraDAOsToListenTo) {
 
   @Override
   public boolean check(foam.core.X x, String permission) {
-    if ( x == null || permission == null ) return false;
+    if ( x == null || permission == null ) {
+      StdoutLogger.instance().debug(getClass().getSimpleName(), "check", "x or permission not provided", permission);
+      return false;
+    }
 
     Permission p = new AuthPermission(permission);
 
     User user = getUserFromContext(x);
     Map<String, Boolean> map = getPermissionMap(user);
 
-    if ( map.containsKey(p.getName()) ) return map.get(p.getName());
+    Logger logger = Loggers.logger(x, this);
 
-    boolean permissionCheck = getDelegate().check(x, permission);
+    Boolean permissionCheck = map.get(p.getName());
+    if ( permissionCheck != null ) {
+      logger.debug("Found cached permission check result", user, p.getName(), permissionCheck);
+      return map.get(p.getName());
```

Comment:
Cant you just return permissionCheck rather than map.get again

Reply:
thanks, missed that